pkg/minikube/machine/cache_images.go

```diff
@@ -207,12 +215,26 @@ func LoadFromCacheBlocking(cmd bootstrapper.CommandRunner, src string) error {
 		return errors.Wrap(err, "transferring cached image")
 	}
 
-	dockerLoadCmd := "docker load -i " + dst
+	var dockerLoadCmd string
+	crio := k8s.ContainerRuntime == constants.CrioRuntime || k8s.ContainerRuntime == constants.Cri_oRuntime
+	if crio {
+		dockerLoadCmd = "sudo podman load -i " + dst
+	} else {
+		dockerLoadCmd = "docker load -i " + dst
+	}
+
+	if crio {
+		podmanLoad.Lock()
```

Comment:
What type of failure is this lock trying to prevent?

Unless it's something specifically terrible, I'd prefer as little state as possible here.

Reply:
As per the commit message, it was running out of memory in the VM when running all the "load" commands in parallel. With docker, things are serialized / queued in the docker daemon. But with podman, it will actually try to run all of them at once. So I had to introduce a lock, for the command to succeed. It's mostly I/O-bound anyway, so you don't lose too much time by it.